storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -1505,35 +1508,36 @@ public static StormTopology addVersions(StormTopology topology) {
         Map<String, Object> defaultsConf = null;
         Map<String, Object> stormConf = null;
         for (String part: cp) {
-            File f = new File(part);
-            if (f.isDirectory()) {
-                if (defaultsConf == null) {
-                    defaultsConf = readConfIgnoreNotFound(yaml, new File(f, "defaults.yaml"));
-                }
-                
-                if (stormConf == null) {
-                    stormConf = readConfIgnoreNotFound(yaml, new File(f, "storm.yaml"));
+            String wildcardSuffix = File.separator + "*";
+            if (part.endsWith(wildcardSuffix)) {
+                // wildcard is given
+                // in java classpath, '*' is translated to '*.jar'
+                File dir = new File(part.substring(0, part.length() - wildcardSuffix.length()));
+                File[] jarFiles = dir.listFiles((dir1, name) -> name.endsWith(".jar"));
+
+                if (jarFiles != null) {
```

Comment:
Nit: Isn't there something really weird going on if this is null? The javadoc only mentions that this can be null if there's an IO error or the dir isn't a directory.

Reply:
I'm not sure. If jarFiles is null, it looks like we'll end up returning an empty config, so I think downstream code will probably break. If that's the case, I'd prefer throwing an exception here so it's obvious what went wrong. Whether we want to throw a custom exception or just remove the null check and get an NPE in the following line makes no difference to me.

Reply:
Yes I also feel better to throw an exception. Throwing exception will not break Nimbus instance too.